### PR TITLE
2 add LSOA col to EPC

### LINKED ISSUE #2: add LSOA column to EPC dataset
Join LSOA/MSOA col to EPC dataset from ONS postcode directory.

### PR DESCRIPTION
**Pull request by crispy-wonton**
Fixes #2 

# Description

Create basic pipeline and add functions required to generate enhanced EPC dataset with LSOA column joined from ONS postcode directory.
- add `base_getters.py`: base getters to get dataset from ZIP files
- add `get_datasets.py`: specific getters to get ONSPD dataset
- add `enhance_epc_functions.py`: functions to clean and prepare EPC postcode column and join EPC with ONSPD to add LSOA column
- add `run_script.py`: run script to generate enhanced EPC dataset with additional LSOA and MSOA columns
- add `requirements.txt`
- add params to config `base.yaml`
- add `schemas.py`: data schemas for importing datasets

# Instructions for Reviewer

In order to test the code in this PR you need to:

- clone repo and follow set up instructions in README
- `cd` to repo folder
- run: `python asf_heat_pump_suitability/pipeline/run_script.py --epc_path s3://asf-daps/lakehouse/processed/epc/deduplicated/processed_dedupl-0.parquet --save_output s3://asf-heat-pump-suitability/outputs/test.parquet`
- This should generate the `test.parquet` file in the above S3 location. The dataset in this file should be EPC (selected cols) + 5 cols from ONSPD: pcd, lsoa11, lsoa21, msoa11, msoa21.

You can test the same function in a notebook with this code:
```
from asf_heat_pump_suitability.pipeline import run_script
epc_path = "s3://asf-daps/lakehouse/processed/epc/deduplicated/processed_dedupl-0.parquet"
output_path = "s3://asf-heat-pump-suitability/outputs/test.parquet"
test = run_script.main(epc_path=epc_path, save_output=output_path)
```

In your review, please could you:
- check the code works for you
- general check for errors/bugs/mistakes in the code and in the logic
- check the output of the code is as expected and desired

# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [ ] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+polars
+s3fs>=2023.3.0
```
Comment (lizgzil) [inline requirements.txt:2]: I got some errors with running `make install`. 

I'm not sure why though! I had to do:
```
pip install polars-lts-cpu
pip install s3fs
```
and then it worked 🤷‍♀️

**Review by lizgzil — COMMENTED** [review]
This all looks great @crispy-wonton ! The script ran perfectly and the output file looked fine to me! (I noticed lots of empty strings in the address2, but I don't think this needs fixing here - I guess later down the line you will do some cleaning steps).

I had a bit of an issue with the environment - not quite sure what was up - but I found a fix (see my comment).

Two things that'd be good to perhaps change:
1. File structure. I'm thinking as this project develops the pipeline folder is going to get quite full - so perhaps this work would be good to be put into a subfolder of pipeline (e.g. "epc_preprocessing"?)
2. it'd also be great to start a pipeline README to keep a log of all these steps/commands to run.

**Comment by crispy-wonton** [comment]
@lizgzil I have added a README in the pipeline folder in branch [4_subset_epc](https://github.com/nestauk/asf_heat_pump_suitability/pull/5). I have created a [separate issue](https://github.com/nestauk/asf_heat_pump_suitability/issues/6) for the restructuring and will do that in a separate branch after merging these two to avoid messy merge conflicts. Are you happy for me to merge this PR with that in mind?

**Comment by lizgzil** [comment]
yep - thanks Roisin!

**Review by lizgzil — APPROVED** [review]

**Pull request merged**